.github/workflows/check-and-build.yml

```diff
@@ -1,15 +1,11 @@
-name: CI
+name: Check and build
 
 on:
-  push:
-    tags:
-      - '*'
   pull_request:
```

Comment:
```suggestion
  push:
    branches:
      - develop
  pull_request:
```

Reply:
Why? The action already that runs on develop already includes this action, this would run it twice?

Reply:
You mean `.github/workflows/populate-gradle-build-cache.yml`? It has been deleted in this PR.

Reply:
No, I meant the `deploy-test` has `uses: ./.github/workflows/check-and-build.yml` and runs on develop, but it actually doesn't 😅 
I'll fix that, thanks